compiler/rustc_mir_build/src/build/mod.rs

```diff
@@ -57,6 +57,8 @@ fn mir_build(tcx: TyCtxt<'_>, def: ty::WithOptConstParam<LocalDefId>) -> Body<'_
             }
```

Comment:
It's a total nit, but I'd prefer to move the `if tcx.sess.opts...` into the query, and unconditionally invoke the query from `mir_build`.